src/core/CMakeLists.txt

```diff
@@ -0,0 +1,45 @@
+
+set(CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS};-O3 -arch=sm_20)#arch=compute_20,code=sm_20)
```

Comment:
Use `--arch=sm_20 --arch=sm_30` and get rid of the comment. That way, code is optimized for both 2.0 and 3.0 GPUs.

Also, don't put `-O3` here. _nvcc_ should use the global optimization level. I think it automatically gets called with `${CMAKE_CXX_FLAGS}`, so you don't need to add it to `CUDA_NVCC_FLAGS`. However, when compiling in debug mode (i.e. `${CMAKE_BUILD_TYPE}` is set to _Debug_), the flag `-G` should be added to `CUDA_NVCC_FLAGS` so that debugging information is generated for GPU code as well.